metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/enrichment/EnricherProvider.java

```diff
@@ -22,99 +23,105 @@
-      }
-    } else if (StringUtils.isNotBlank(enrichmentUrl)) {
-      try {
-        entityResolver = new RemoteEntityResolver(new URL(enrichmentUrl), batchSizeEnrichment,
-                createRestTemplate());
-      } catch (MalformedURLException e) {
-        LOGGER.debug("There was a problem creating entity resolver");
-        throw new EnrichmentException("There was a problem while creating new Enricher", e);
-      }
-    } else {
-      throw new EnrichmentException("We must have either a non-null entity resolver creator,"
-              + " or a non-blank enrichment URL.", null);
+    private RecordParser recordParser;
+    private EntityResolverCreator entityResolverCreator;
+    private String entityManagementUrl;
+    private String entityApiUrl;
+    private String entityApiKey;
+
+    /**
+     * Set the record parser to use. The default is null, in which case an instance of {@link
+     * MetisRecordParser} will be used.
+     *
+     * @param recordParser The record parser to use.
+     */
+    public void setRecordParser(RecordParser recordParser) {
+        this.recordParser = recordParser;
     }
 
-    // Done.
-    return new EnricherImpl(recordParser == null ? new MetisRecordParser() : recordParser,
-            entityResolver, new EntityMergeEngine());
-  }
+    /**
+     * Convenience method for {@link #setEntityResolverCreator(EntityResolverCreator)} that uses the
+     * passed entity resolver instead of creating one on demand. Calling this method counts as calling
+     * {@link #setEntityResolverCreator(EntityResolverCreator)} with a non-null creator.
+     *
+     * @param entityResolver The entity resolver to use.
+     */
+    public void setEntityResolver(EntityResolver entityResolver) {
+        setEntityResolverCreator(() -> entityResolver);
+    }
 
-  /**
-   * Implementations of this interface create instances of {@link EntityResolver}.
-   */
-  @FunctionalInterface
-  public interface EntityResolverCreator {
+    /**
+     * Set the entity resolver creator to use. The default is null, in which case a {@link
+     * ClientEntityResolver} will be used with the connection settings in this class, and {@link
+     * #setPropertiesValues(String, String, String)} will need to have been called.
+     *
+     * @param entityResolverCreator A creator for the entity resolver.
+     */
+    public void setEntityResolverCreator(EntityResolverCreator entityResolverCreator) {
+        this.entityResolverCreator = entityResolverCreator;
+    }
 
     /**
-     * Creates an instance of {@link EntityResolver}.
+     * Set the properties values of the enrichment service. The default is null, in which case {@link
+     * #setEntityResolverCreator(EntityResolverCreator)} will need to have been called.
      *
-     * @return An instance of {@link EntityResolver}.
-     * @throws EnrichmentException In case there was a problem creating the instance.
+     * @param entityManagementUrl The url of the entity service
```

Comment:
same here _This is the URL of the entity management service._

Reply:
Done